### PR TITLE
[Reland]Change ATEN generator argument type to const std::optional<Generator>&

### PR DESCRIPTION
**Pull request by cyyever**
Reland of #6686 because the upstream can be merged.

### CODE REVIEW

**Comment by cyyever** [comment]
@alanwaketan Need your approval once the tests pass since the upstream PR has been merged.

**Review by alanwaketan — APPROVED** [review]
LGTM. Please remove the .torch_pin before landing.

**Comment by cyyever** [comment]
@alanwaketan torch_mp_op test failure seems unrelated.

**Pull request merged**
